File: server/engines/lua/defaults/defaults-hl.lua
-- Sample whiteboard: Lua HiLo game
guesses = 0
guess = ""

io.write("Guess a number between 1 and 20")

math.randomseed(os.time())
secret = math.floor(math.random(1, 20));

repeat
    guesses = guesses + 1
    io.flush()
    guess = tonumber(io.read())
    
    if guess == nil then
        io.write("Try a number...")
    elseif guess < secret then
        io.write("Try higher...")
    elseif guess > secret then
        io.write("Try lower...")
    end
until guess == secret

io.write("Yes! You guessed it in "..guesses.." guesses.\n")

File: server/engines/ocaml/defaults/defaults-hl.ml
(* Sample whiteboard: OCaml HiLo game *)
Random.self_init ();;

let rec hilo n guesses =
    let i = read_int () in 
        if i = n then 
            Printf.printf "Yes! You guessed it in %i guesses." guesses
        else 
            let () = 
                if i < n then 
                    Printf.printf "Try higher..."
                else if i > n then
                    Printf.printf "Try lower..."
                else Printf.printf "Try a number..."
            in hilo n (guesses + 1) ;;

Printf.printf "Guess a number between 1 and 20";;          
hilo ((Random.int 19) + 1) 1;;